karapace/avro_compatibility.py

```diff
@@ -72,6 +94,15 @@ def incompatible(incompat_type: SchemaIncompatibilityType, message: str, locatio
         )
         return ret
 
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, SchemaCompatibilityResult):
+            return False
+
+        return (
+            self.locations == other.locations and self.messages == other.messages
+            and self.compatibility == other.compatibility and self.incompatibilities == other.incompatibilities
+        )
```

Comment:
In in the [Java version](https://github.com/apache/avro/blob/eda435931b8dbd913b98567814494ddfe4bea016/lang/java/avro/src/main/java/org/apache/avro/SchemaCompatibility.java#L610), the class doesn't have `locations` and `messages` and hence they are not used for comparison. I wonder what might be the implications of including them into comparison in our implementation.

Reply:
AFAIK `locations` and `messages` were added to aid debugging. What could happen is the tests is that if the code is not deterministic we would see more failures with our code, e.g. if the messages themselves change across runs because a timestamp is added, then we would see more failures. For testing ATM this seems fine though.